debian/patches/0007-add-bt2390-eetf-and-code-refactor-to-opencl-tonemap.patch

```diff
@@ -1644,7 +1652,36 @@ Index: FFmpeg/libavfilter/vf_tonemap_opencl.c
      if (ctx->range_in == AVCOL_RANGE_JPEG)
          av_bprintf(&header, "#define FULL_RANGE_IN\n");
  
-@@ -199,19 +538,41 @@ static int tonemap_opencl_init(AVFilterC
+     if (ctx->range_out == AVCOL_RANGE_JPEG)
+         av_bprintf(&header, "#define FULL_RANGE_OUT\n");
+ 
++    if (ctx->in_desc->comp[0].depth == 10) {
++        av_bprintf(&header, "#define INPUT_QUANTIZATION_OFFSET 0.0009765774014f\n");
++        av_bprintf(&header, "#define INPUT_Y_SCALE 1.1678082192f\n");
++        av_bprintf(&header, "#define INPUT_UV_SCALE 1.1417410714f\n");
++    } else if (ctx->in_desc->comp[0].depth == 16) {
++        // 16bit texture is actually 12bit
++        av_bprintf(&header, "#define INPUT_QUANTIZATION_OFFSET 0.0002441443503f\n");
++        av_bprintf(&header, "#define INPUT_Y_SCALE 1.1689497717f\n");
++        av_bprintf(&header, "#define INPUT_UV_SCALE 1.1428571429f\n");
++    } else {
++        // should not happen, but for completeness
++        av_bprintf(&header, "#define INPUT_QUANTIZATION_OFFSET 0.0f\n");
++        av_bprintf(&header, "#define INPUT_Y_SCALE 1.0f\n");
++        av_bprintf(&header, "#define INPUT_UV_SCALE 1.0f\n");
++    }
++
++    if (ctx->out_desc->comp[0].depth > 8) {
++        av_bprintf(&header, "#define RESCALE_LIMITED_RANGE_OUTPUT\n");
++    }
++
++    if (ctx->in_desc->comp[0].depth == 10)
++        av_bprintf(&header, "#define OUTPUT_QUANTIZATION_OFFSET 0.0009765774014f\n");
```

Comment:
`(ctx->in_desc->comp[0].depth == 10)`

in_desc => out_desc ?

`> 10` needs to be considered?

Reply:
Fixed for out_desc. We don't consider > 10 for now and assume it always uses full 16bit, which does not need such offset.